troubleshoot_data.py

```diff
@@ -0,0 +1,432 @@
+    a successfully published data package.
+    This also prints if there are any missing or unexpected results.
+
+    :param ctx:          Combined type of a callback and rei struct
+    :param data_package: String representing the data package collection path.
+    :param write_stdout: A boolean representing whether to write to stdout or rodsLog
+
+    :returns:            A 2-tuple containing boolean results of checking results
+    """
+    extracted_avus = avu.of_coll(ctx, data_package)
+    results = misc.check_data_package_system_avus(extracted_avus)
+
+    if not results["no_missing_avus"]:
+        log.write(ctx, "check_data_package_system_avus: There are some missing AVUs in data package <{}> - {}".format(data_package, list(results["missing_avus"])), write_stdout)
+
+    if not results["no_unexpected_avus"]:
+        log.write(ctx, "check_data_package_system_avus: There are some unexpected AVUs in data package <{}> - {}".format(data_package, list(results["unexpected_avus"])), write_stdout)
+
+    return (results["no_missing_avus"], results["no_unexpected_avus"])
+
+
+def check_datacite_doi_registration(ctx, data_package, offline, write_stdout):
+    """
+    Check the registration status of both versionDOI and baseDOI with the DataCite API,
+    ensuring that both DOIs return a 200 status code, which indicates successful registration.
+
+    :param ctx:          Combined type of a callback and rei struct
+    :param data_package: String representing the data package collection path.
+    :param offline:      Whether to not connect to datacite
+    :param write_stdout: A boolean representing whether to write to stdout or rodsLog
+
+    :returns:            A tuple of booleans indicating check success or not.
+    """
+    if offline:
+        return True, True
+
+    version_doi_check = False
+    base_doi_check = False
+
+    try:
+        version_doi = get_val_for_attr_with_pub_prefix(ctx, data_package, "versionDOI")
+        status_code = datacite.metadata_get(ctx, version_doi)
+        version_doi_check = status_code == 200
+    except ValueError as e:
+        log.write(ctx, "check_datacite_doi_registration: Error while trying to get versionDOI - {}".format(e), write_stdout)
+
+    previous_version = ''
+    try:
+        previous_version = get_val_for_attr_with_pub_prefix(ctx, data_package, "previous_version")
+    except Exception:
+        pass
+
+    if previous_version:
+        try:
+            base_doi = get_val_for_attr_with_pub_prefix(ctx, data_package, "baseDOI")
+            status_code = datacite.metadata_get(ctx, base_doi)
+            base_doi_check = status_code == 200
+        except ValueError as e:
+            log.write(ctx, "check_datacite_doi_registration: Error while trying to get baseDOI - {}".format(e), write_stdout)
+
+    return (version_doi_check, base_doi_check)
+
+
+def get_val_for_attr_with_pub_prefix(ctx, data_package, attribute_suffix):
```

Comment:
I'm not super happy with this name, happy for suggestions (or go back to `get_attribute_value`?).